akka-cluster-tools/src/main/scala/akka/cluster/singleton/ClusterSingletonManager.scala

```diff
@@ -761,6 +778,15 @@ class ClusterSingletonManager(
       sender() ! Done // reply to ask
       stay
 
+    case Event(MemberDowned(m), OldestData(singleton, singletonTerminated)) if m.uniqueAddress == cluster.selfUniqueAddress ⇒
+      if (singletonTerminated) {
+        logInfo("Self downed, stopping ClusterSingletonManager")
+        stop()
+      } else {
+        logInfo("Self downed, stopping")
+        gotoStopping(singleton)
+      }
+
```

Comment:
I'm trying to remember about the handover stuff, where it tries to handover as long as remoting is still connected, does this earlier stop interfere with that in any way?

Reply:
Handover is intended for the leaving scenario, when downing you shouldn't expect a smooth/quick handover. Normally it can't talk to the downed node anyway because it is on other side of network partition or is crashed.

Reply:
Ah, ofc